### PR TITLE
Iterate on design & add PWA support

### PR DESCRIPTION
**Pull request by brettchalupa**
This iterates on the UI to make it a bit more user friendly in a variety of ways. I focused on keeping the UI simple, accessible, and semantic.

I specifically didn't change any JS and tried to not change anything relating to the core functionality. It's mostly CSS and some HTML adjustments.

See commits for details, I tried to make them small and atomic.

In addition to using some brand colors and styles, I've also added the logo, expanded on the README, and made it so that the website can be installed as a Progressive Web App to make it possible for Chrome users to install on their desktop for easy access.

## Screenie

screenshot of the LED Matrix Input Module control

### Progressive Web App Install UI

image

## Acceptance Criteria

1. Check out this PR - `gh pr checkout 6`
2. Boot up a web server - if you have Python, `python3 -m http.server` or with Ruby: `ruby -run -e httpd . -p 5555`
3. View the site locally and ensure it:
   1. looks nice
   2. works with input modules if ya' got em

Happy to iterate on this and all feedback welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+Opera, etc.).
 
-Then use the connect button to connect. Drawing on the screen should
-automatically update the LED matrix.
+To start, simply open `index.html`. There is no server to run. If you want, you
+can boot a simple web server with Python: `python3 -m http.server`
 
-Click and drag to draw, CTRL+click to erase.
+Use the "Connect Left" and "Connect Right" buttons to connect to each
+respective input module. Draw on each input browser in your browser, and it
```
Comment (zachfeldman) [inline README.md]: Nit: input browser -> input module?

File: README.md [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
 
-###### Dot Matrix Tool
+## Dot Matrix Tool
 
 This repository is based on code from https://github.com/stefangordon/dotmatrixtool.
 See also http://dotmatrixtool.com
+
+## Contributing
+
+Contrirbutions are welcome. Submit pull requests to the `usbserial` branch.
```
Comment (zachfeldman) [inline README.md]: Contrirbutions -> Contributions

File: index.html [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
-    <meta name="description" content="An online tool for controlling the Framework LED Matrix">
+    <meta name="description" content="An web tool for controlling the Framework LED Matrix input module">
```
Comment (zachfeldman) [inline index.html]: An -> A

Comment (zachfeldman) [reply index.html]: Also maybe we want to add "for the Framework Laptop 16" to the end?

File: manifest.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (zachfeldman) [inline manifest.json:1]: Very cool idea, love it! FYI it only worked when running it with the Ruby server. It'll probably work better on https://ledmatrix.frame.work/.

File: README.md [hunk 5]
```diff
@@ -31,3 +31,8 @@ See also http://dotmatrixtool.com
 ## Contributing
 
 Contributions are welcome. Submit pull requests to the `usbserial` branch.
+
+When preparing your pull request, please be sure:
+
+1. the tool still works with connected LED Matrix Input Modules
```
Comment (zachfeldman) [inline README.md:43]: Nit: the -> The

File: app.js [hunk 6]
```diff
@@ -116,8 +116,8 @@ function updateTableLeft() {
 	var height = matrix_left.length;
 
   $table_left = populateTable(null, height, width, "left");
-	$('#_grid_left').html('');
-	$('#_grid_left').append($table_left);
+	$('#led-grid_left').html('');
```
Comment (zachfeldman) [inline app.js:119]: Nice and clean :shower:

**Review by zachfeldman — APPROVED** [review]
Thanks @brettchalupa , this is a huge improvement!

It is working well locally with my LED matrix. Only issue is that after some time I saw this JS error, even when refreshing the page. May need some help from @JohnAZoidberg to fix:
image
it did work again when I tried it with the PWA! So I bet it would work again if I restarted Chrome.

We may want to give credit to the original person that created this tool on the page, I think that would be cool - see the README "This repository is based on code from https://github.com/stefangordon/dotmatrixtool."

File: README.md (hunk 2 above)
Comment (zachfeldman) [reply README.md]: Whoops you already fixed this!

File: manifest.json (hunk 4 above)
Comment (brettchalupa) [reply manifest.json:1]: Yeah, I'd assume that's a limitation of how PWAs work. If someone downloads the files and uses the download, I think it's safe to assume they don't need the PWA installed then.

File: README.md (hunk 5 above)
Comment (brettchalupa) [reply README.md:43]: I prefer it being lower case since they're not proper sentences

**Comment by brettchalupa** [comment]
new screenie w/ hacker vibes bringing back the monospace font

image

**Comment by brettchalupa** [comment]
@zachfeldman thanks for all of the feedback, I believe I've addressed most of the issue and implemented the suggestions. 🙌 

Could you log a separate GitHub issue in this repo for the console error? That seems unrelated to the changes and isn't something I can fix yet (you or @JohnAZoidberg could investigate though if you're feeling interested 😸 ).

**Comment by brettchalupa** [comment]
I'm going to rebase this on top of the latest in `usbserial` so that it gets the Cloudflare Pages preview deployment for easy review + fixes some conflicts I introduced.

**Comment by brettchalupa** [comment]
yay the preview deploys are working now, that's legit! you can see them as part of the PR's deployments if you want to give it a test on a hosted version

**Review by zachfeldman — APPROVED** [review]
Thanks for addressing all of my comments @brettchalupa - still working great with my LED Matrices!

**Pull request merged**